packages/eas-cli/src/build/build.ts

```diff
@@ -174,8 +175,28 @@ export async function prepareBuildRequestForPlatformAsync<
   }
   assert(projectArchive);
 
-  const runtimeMetadata = await createAndMaybeUploadFingerprintAsync(ctx);
-  const metadata = await collectMetadataAsync(ctx, runtimeMetadata);
+  let runtimeAndFingerprintMetadata: {
+    runtimeVersion?: string | undefined;
+    fingerprintHash?: string | undefined;
+    fingerprintSource?: FingerprintSource | undefined;
+    fingerprint?: {
+      fingerprintSources: object[];
+      isDebugFingerprintSource: boolean;
+    };
+  } = await computeAndMaybeUploadFingerprintFromExpoUpdatesAsync(ctx);
+  if (!runtimeAndFingerprintMetadata?.fingerprint) {
+    const fingerprint = await computeAndMaybeUploadFingerprintWithoutExpoUpdatesAsync(ctx);
+    runtimeAndFingerprintMetadata = {
+      ...runtimeAndFingerprintMetadata,
+      ...fingerprint,
+    };
+  } else {
+    runtimeAndFingerprintMetadata = {
+      ...runtimeAndFingerprintMetadata,
+      fingerprintHash: runtimeAndFingerprintMetadata.runtimeVersion,
+    };
+  }
```

Comment:
nit: maybe we can extract it to the separate function called `computeAndMaybeUploadRuntimeAndFingerprintMetadata`?